tests/kafkatest/utils/util.py

```diff
@@ -30,34 +30,96 @@ def _kafka_jar_versions(proc_string):
         - kafka-streams-1.0.0-SNAPSHOT.jar
         - kafka-streams-0.11.0.0-SNAPSHOT.jar
     """
+    versions=[]
 
+    #
+    # 5 kafka_#.#-*-SNAPSHOT.jar possibilities
+    #
+    # 2 possible kafka_#.#-#.#.#[.#]-SNAPSHOT.jar matches:
     # Pattern example: kafka_2.11-1.0.0-SNAPSHOT.jar (we have to be careful not to partially match the 4 segment version string)
-    versions = re.findall("kafka_[0-9]+\.[0-9]+-([0-9]+\.[0-9]+\.[0-9]+)[\.-][a-zA-z]", proc_string)
-
+    versions.extend(re.findall("kafka_[0-9]+\.[0-9]+-([0-9]+\.[0-9]+\.[0-9]+)-SNAPSHOT\.jar", proc_string))
```

Comment:
```suggestion
    versions.extend(re.findall(r"kafka_[0-9]+\.[0-9]+-([0-9]+\.[0-9]+\.[0-9]+)-SNAPSHOT\.jar", proc_string))
```
Use raw strings to prevent parsing the escaped `.` or use double backslashes